src/GraphQL/DI/DefaultServiceProvider.cs

```diff
@@ -11,15 +11,22 @@ namespace GraphQL
     public sealed class DefaultServiceProvider : IServiceProvider
     {
         /// <summary>
-        /// Gets an instance of the specified type. Can not return <see langword="null"/> but may throw exception.
+        /// Gets an instance of the specified type. Returns <see langword="null"/> for interfaces.
+        /// Can not return <see langword="null"/> for classes but may throw exception.
         /// </summary>
         /// <param name="serviceType">Desired type</param>
         /// <returns>An instance of <paramref name="serviceType"/>.</returns>
-        public object GetService(Type serviceType)
+        public object? GetService(Type serviceType)
         {
             if (serviceType == null)
                 throw new ArgumentNullException(nameof(serviceType));
 
+            if (serviceType == typeof(IServiceProvider))
+                return this;
+
+            if (serviceType.IsInterface)
+                return null;
```

Comment:
Why??? What if `serviceType` is abstract class? It wil fail to be created and we get throw on line 36. I would also throw on interfaces:
```c#
  public object? GetService(Type serviceType)
        {
            if (serviceType == null)
                throw new ArgumentNullException(nameof(serviceType));

      if (serviceType.IsInterface || serviceType.IsAbstract)
                throw new ArgumentException(nameof(serviceType), "This provider can create only concrete classes.");
}
```

Reply:
The same check for `static` classes may be added.

Reply:
OK, I understood that it's fine to return `null` for interfaces.

Reply:
So you may also return `null` for abstract and static classes.

Reply:
Right. I thought of abstract but didn’t think of static. Good catch.

Reply:
Correction, apparently I didn’t think of abstract here. Must be thinking of somewhere else. Doubly good catch

Reply:
No property IsStatic, but static classes are marked as abstract, so only one more check needed.

Reply:
Yes. Static = abstract + sealed.

Reply:
https://github.com/graphql-dotnet/graphql-dotnet/pull/2769